### PR TITLE
Universal test for RangeTombstoneSnapshotMigrateFromLast, refactor

### PR DESCRIPTION
**Pull request by pdillinger**
Summary:
* Expand RangeTombstoneSnapshotMigrateFromLast in tiered_compaction_test (originally from #13124) to reproduce a failure in universal compaciton (as well as leveled), when a specific part of the test is uncommented.
* Small refactoring to eliminate unnecessary fields in SubcompactionState. Adding a bool parameter to SubcompactionState::AddToOutput here will make more sense in the next PR (which I'm trying to keep
from getting too big).
* Improve debuggability and performance of some other tests
* Remove accidentally committed test "BlahPrecludeLastLevel" which was a temporary copy of CompactionServiceTest.PrecludeLastLevel

Test Plan: existing tests, updated/expanded tests

### CODE REVIEW

File: db/compaction/tiered_compaction_test.cc [hunk 1]
```diff
@@ -1651,7 +1651,8 @@ TEST_P(PrecludeLastLevelTest, CheckInternalKeyRange) {
 
   ASSERT_EQ("0,0,0,0,0,2,2", FilesPerLevel());
 
-  auto VerifyLogicalState = [&]() {
+  auto VerifyLogicalState = [&](int line) {
+    SCOPED_TRACE("Called from line " + std::to_string(line));
```
Comment (cbi42) [inline db/compaction/tiered_compaction_test.cc:1655]: 👍

File: db/compaction/tiered_compaction_test.cc [hunk 2]
```diff
@@ -1732,10 +1755,16 @@ TEST_P(PrecludeLastLevelTest, RangeTombstoneSnapshotMigrateFromLast) {
       [&] { mock_clock_->MockSleepForSeconds(static_cast<int>(1000)); });
 
   ASSERT_OK(Flush());
-  MoveFilesToLevel(5);
+
+  // Send three files to next-to-last level (if explicitly needed)
+  if (!universal) {
+    MoveFilesToLevel(5);
+  }
+
```
Comment (cbi42) [inline db/compaction/tiered_compaction_test.cc:1763]: Need to wait for compaction here?

**Review by cbi42 — APPROVED** [review]
LGTM

File: db/compaction/tiered_compaction_test.cc (hunk 2 above)
Comment (pdillinger) [reply db/compaction/tiered_compaction_test.cc:1763]: Yes, about to submit that.

**Pull request closed without merging**